external_resources_io/terraform/generators.py

```diff
@@ -0,0 +1,208 @@
+    }
+
+
+def _generate_terraform_variables_from_model(model: type[BaseModel]) -> dict:
+    """Generates Terraform variables json."""
+    return {"variable": _generate_fields(model)}
+
+
+def _generate_terraform_variable(python_type: Any, default: Any = None) -> dict:
+    """Generates a Terraform variable block."""
+    variable_block: dict[str, str | None] = {"type": _get_terraform_type(python_type)}
+
+    if default is not PydanticUndefined:
+        variable_block["default"] = (
+            default.model_dump()
+            if default and isinstance(default, BaseModel)
+            else default
+        )
+    return variable_block
+
+
+def _convert_generic_types(origin: Any, args: Any) -> str:  # noqa: PLR0911
+    """Convert generic types to Terraform types."""
+    match origin:
+        case t if t in {list, Sequence}:
+            return f"list({_get_terraform_type(args[0])})" if args else "list(any)"
+
+        case t if t is set:
+            return f"set({_get_terraform_type(args[0])})" if args else "set(any)"
+
+        case t if t is dict:
+            return f"map({_get_terraform_type(args[1])})" if args else "map(any)"
+
+        case t if t is Literal:
+            return "string"
+
+        case t if t in {UnionType, Union}:
+            if type(None) in args:
+                return _get_terraform_type(args[0])
+            return "any"
+
+        case _:
+            return "any"
+
+
+def _convert_basic_types(python_type: Any) -> str:  # noqa: PLR0911
+    """Convert basic python types to Terraform types."""
+    match python_type:
+        case t if t is str:
+            return "string"
+        case t if t is int:
+            return "number"
+        case t if t is bool:
+            return "bool"
+        case t if t is list:
+            return "list(any)"
+        case t if t is set:
+            return "set(any)"
+        case t if t is dict:
+            return "map(any)"
+        case t if issubclass(t, BaseModel):
+            # nested model
+            fields_types = ",".join(
+                f"{k} = {v['type']}" for k, v in _generate_fields(t).items()
+            )
+            return f"object({{{fields_types}}})"
+        case _:
+            return "any"
+
+
+def _get_terraform_type(python_type: Any) -> str:
+    """Maps Python types to Terraform types using structural pattern matching."""
+    origin = get_origin(python_type)
+    args = get_args(python_type)
+
+    return (
+        _convert_generic_types(origin, args)
+        if origin is not None
+        else _convert_basic_types(python_type)
+    )
+
+
+def _convert_json_value_to_hcl(value: Any) -> str:  # noqa: PLR0911
+    """Converts a JSON value to HCL."""
+    match value:
+        case t if isinstance(t, str):
+            return f'"{value}"'
+        case t if isinstance(t, bool):
+            return str(value).lower()
+        case t if isinstance(t, int | float):
+            return str(value)
+        case t if isinstance(t, list):
```

Comment:
`value` can also be `set`, right? The actual input `value` is not only json types, but all python types.

Reply:
Fixed